src/sources/kubernetes_logs/mod.rs

```diff
@@ -166,6 +174,31 @@ pub struct Config {
     log_namespace: Option<bool>,
 }
 
+/// File position to use when reading a new file.
+#[configurable_component]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum ReadFromConfig {
+    /// Read from the beginning of the file.
+    Beginning,
+
+    /// Start reading from the current end of the file.
+    End,
+}
+
+impl From<ReadFromConfig> for ReadFrom {
+    fn from(rfc: ReadFromConfig) -> Self {
+        match rfc {
+            ReadFromConfig::Beginning => ReadFrom::Beginning,
+            ReadFromConfig::End => ReadFrom::End,
+        }
+    }
+}
```

Comment:
Since this is duplicated from the `file` source, I'm thinking it should be moved to a shared location - maybe into the `file_server` itself?

Reply:
I've tried to do it but had some strange issues with the build - the compiler failed to compile `configurable_component` if I moved it to `file_server` file.

Reply:
🤔 I'll see if I can take a quick look at that this afternoon, might be an easy fix. We definitely have some `configurable(derived)`'s around that refer into `lib/` crates.

Reply:
At a glance before I leave for the weekend, it looks like we'd just need to add a couple deps:

```toml
vector-config = { path = "../vector-config", default-features = false }
vector-config-common = { path = "../vector-config-common", default-features = false }
vector-config-macros = { path = "../vector-config-macros", default-features = false }
```

I dropped the snippet just into `src/lib.rs` and `use`'d `vector_config::configurable_component` and it seemed to compile without issue.

Reply:
Thanks! It works! I thought adding these dependencies could violate somehow Vector's policies regarding dependencies but I see that it's okay :)

Reply:
Done